packages/dev/core/src/Maths/math.color.ts

```diff
@@ -135,58 +147,198 @@ export class Color3 {
      * @param otherColor defines the second operand
      * @returns the new Color3 object
      */
-    public multiply(otherColor: DeepImmutable<Color3>): Color3 {
-        return new Color3(this.r * otherColor.r, this.g * otherColor.g, this.b * otherColor.b);
+    public multiply(otherColor: DeepImmutable<this>): this {
+        return new (this.constructor as Constructor<typeof Color3, this>)(this.r * otherColor.r, this.g * otherColor.g, this.b * otherColor.b);
     }
 
     /**
      * Multiply the rgb values of the Color3 and the given Color3 and stores the result in the object "result"
      * @param otherColor defines the second operand
      * @param result defines the Color3 object where to store the result
-     * @returns the current Color3
+     * @returns the result Color3
      */
-    public multiplyToRef(otherColor: DeepImmutable<Color3>, result: Color3): Color3 {
+    public multiplyToRef<T extends this>(otherColor: DeepImmutable<this>, result: T): T {
         result.r = this.r * otherColor.r;
         result.g = this.g * otherColor.g;
         result.b = this.b * otherColor.b;
+        return result;
+    }
+
+    /**
+     * Multiplies the current Color3 coordinates by the given ones
+     * @param otherColor defines the second operand
+     * @returns the current updated Color3
+     */
+    public multiplyInPlace(otherColor: DeepImmutable<Color3>): this {
+        this.r *= otherColor.r;
+        this.g *= otherColor.g;
+        this.b *= otherColor.b;
+        return this;
+    }
+
+    /**
+     * Returns a new Color3 set with the result of the multiplication of the current Color3 coordinates by the given floats
+     * @param r defines the r coordinate of the operand
+     * @param g defines the g coordinate of the operand
+     * @param b defines the b coordinate of the operand
+     * @returns the new Color3
+     */
+    public multiplyByFloats(r: number, g: number, b: number): this {
+        return new (this.constructor as Constructor<typeof Color3, this>)(this.r * r, this.g * g, this.b * b);
+    }
+
+    /* Do not use */

```

Comment:
You should add the `@internal` tag so that this function (and the ones below) is not discoverable through intellisense / won't appear in the doc.